File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/Turtle.java
package model.Organisms;

import model.Animal;
import model.Organism;
import model.Point;
import model.World;

import java.awt.*;
import java.util.Objects;

public class Turtle extends Animal {
    private Color turtleColor = Color.GREEN;

    public Turtle(World world, Point position){
        super(world, position, 2, 1);
    }

    @Override
    public Color getColor(){
        return turtleColor;
    }

    @Override
    public String getNazwa(){
        return "Zolw";
    }
    @Override
    public void Action(){
        int randomNum = random.nextInt(100);
        if (randomNum < 75){
            increaseAge();
            world.addEvent("Zółw zostaje w miejscu na pozycji (" + this.getPosition().getX() + ", " + this.getPosition().getY() + ")");
            return;
        }
        super.Action();
    }

    @Override
    public Organism makeNewOrganism(Point position){
        return new Turtle(world, position);
    }

    @Override
    public void Collision(Organism attacker){
        if (Objects.equals(this.getNazwa(), attacker.getNazwa())){
            super.Multiplitaction(attacker);
            return;
        }

        if (attacker.getPower() < 5){
            world.addEvent("Zołw odparł atak " + attacker.getNazwa() + " na pozycji (" + getPosition().getX() + ", " + getPosition().getY() + ")");
            return;
        }

        collisionHelper(attacker, this.getPosition());
    }
}

File: WirtualnySwiatPython/Animals/Turtle.py
import random

from animal import Animal
from config import LIME

class Turtle(Animal):
    def __init__(self, world, position):
        super().__init__(world, position, power=2, initiative=1)
        self.color = LIME
    
    def getNazwa(self):
        return "Zolw"
    
    def getColor(self):
        return self.color

    def makeNewOrganism(self, position):
        return Turtle(self.world, position)
    
    def Action(self):
        ranNum = random.randint(0, 99)
        if ranNum < 75:
            self.increaseAge()
            self.world.addEvent(f"Zolw zostaje w miejscu na pozycji ({self.getPosition().GetX()}, {self.getPosition().GetY()})")
        else:
            super().Action()

    def Collision(self, attacker):
        if self.getNazwa() == attacker.getNazwa():
            self.Multiplication(attacker)
            return

        if attacker.getPower() < 5:
            self.world.addEvent(f"Zolw odparl atak {attacker.getNazwa()} na poztcji ({self.getPosition().GetX()}, {self.getPosition().GetY()})")
            return

        self.collisionHelper(attacker, self.position)